### PR TITLE
IrvineHacks 2025 Preview Page and Initial Setup

### PR DESCRIPTION
**Pull request by waalbert**
- Updates existing site to only show IrvineHacks 2025 Preview Page
- Updates meta titles to "IrvineHacks 2025" or "Page | IrvineHacks 2025"
- Updates dates for when applications open (December 6, 2024 12:00 am) and close (January 10, 2025 11:59 pm) in `hasDeadlinePassed.ts` and our `/apply` FastAPI endpoint
- Creates new utility function `haveApplicationsOpened` to automatically update "Coming Soon" button to the apply button when applications open

### CODE REVIEW

File: apps/site/src/app/(main)/(home)/sections/Landing/Landing.tsx [hunk 1]
```diff
@@ -1,41 +1,37 @@
...
+					IrvineHacks 2025
 				</h1>
-				<p className="font-display text-2xl md:text-3xl">January 26&ndash;28</p>
 				{deadlinePassed ? (
 					<Button className="z-10" text="Applications have closed!" disabled />
-				) : (
+				) : applicationsOpened ? (
 					<Button className="z-10" text="Apply" href="/apply" />
+				) : (
+					<Button className="z-10" text="Coming Soon..." disabled />
```
Comment (Bl20052005) [inline apps/site/src/app/(main)/(home)/sections/Landing/Landing.tsx:34]: note: the hover still appears to be on however you can't click on the button

File: apps/site/src/app/(main)/(home)/sections/Landing/Landing.tsx
Comment (Bl20052005) [inline apps/site/src/app/(main)/(home)/sections/Landing/Landing.tsx:1]: The other website sections (besides landing) are still visible, which makes the current transition of backgrounds slightly jarring but we'll fix that very soon I imagine

**Review by Bl20052005 — APPROVED** [review]

File: apps/site/src/app/(main)/(home)/sections/Landing/Landing.tsx (hunk 1 above)
Comment (waalbert) [reply apps/site/src/app/(main)/(home)/sections/Landing/Landing.tsx:34]: Graphics wanted the hover effect for the buttons, which I left for disabled buttons. I believe this is fine functionality wise.

Comment (Bl20052005) [reply apps/site/src/app/(main)/(home)/sections/Landing/Landing.tsx:34]: yes, functionality is fine

File: apps/site/src/app/(main)/(home)/sections/Landing/Landing.tsx
Comment (waalbert) [reply apps/site/src/app/(main)/(home)/sections/Landing/Landing.tsx:1]: The environment variable `MAINTENANCE_MODE_HOME` has been set up on Vercel, so the rest of the home page is hidden on prod, but on previews we will see the entire home page.

Comment (Bl20052005) [reply apps/site/src/app/(main)/(home)/sections/Landing/Landing.tsx:1]: ah got it

**Review by IanWearsHat — APPROVED** [review]

**Pull request merged**